src/io/open.rs

```diff
@@ -11,33 +11,23 @@ use std::path::Path;
 #[allow(dead_code)]
 pub(crate) struct Open {
     pub(crate) path: CString,
-    pub(crate) flags: libc::c_int,
 }
 
 impl Op<Open> {
     /// Submit a request to open a file.
     pub(crate) fn open(path: &Path, options: &OpenOptions) -> io::Result<Op<Open>> {
-        use io_uring::{opcode, types};
         let path = super::util::cstr(path)?;
-        let flags = libc::O_CLOEXEC
-            | options.access_mode()?
-            | options.creation_mode()?
-            | (options.custom_flags & !libc::O_ACCMODE);
-
+        // Get a reference to the memory. The string will be held by the
+        // operation state and will not be accessed again until the operation
+        // completes.
```

Comment:
```suggestion
        // TODO: SAFETY:
        // how exactly is the safety requirement fulfilled here?
```

Turn formatting to https://rust-lang.github.io/rust-clippy/master/#/undocumented as I could imagine that is expected as the lint has been enabled in pageserver as well.